### ISSUE TITLE
CA5385 - False positive

### ISSUE THREAD

**Issue by elyoh** [dotnet/roslyn-analyzers#2697]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.3 (Latest)
2.9.4-beta1.final

#### Diagnostic ID

CA5385 Asymmetric encryption algorithm RSACryptoServiceProvider's key size is less than 2048. Switch to an RSA with at least 2048 key size, ECDH or ECDSA algorithm instead. **[no documentation yet]**

#### Repro steps

The following code causes a CA5385 warning:

```csharp
var rsa = new RSACryptoServiceProvider();
rsa.FromXmlString(xml);
return rsa;
```
It is unclear why CA5385 is raised because it is not possible for the analyser to know if the supplied xml (or parameters) do indeed use a key size of less than 2048.  For example, the xml may have been generated with an appropriate key length as follows:

```csharp
var rsa = new RSACryptoServiceProvider(4096))
var xml = rsa.ToXmlString(true);
```

#### Expected behavior
It appears from the error description that CA5385 should only be raised if he RSACryptoServiceProvider is initialised with a key size of less than 2048.

The xml or parameters used to initialise the RSACryptoServiceProvider may indeed use a key size of 2048 or above, but there is no way for the analyser to detect that.

At the very least, the description of the error seems to be misleading.

#### Actual behavior
CA5385 is raised.

**Comment by mavasani** [comment]
@dotpaul @LLLXXXCCC

**Comment by dotpaul** [comment]
Thanks for reporting @elyoh!
